csv-generator/tests/consumers_test/manuscript_consumer_test.py

```diff
@@ -32,7 +32,7 @@ def test_will_handle_no_msid(manuscript_consumer: ManuscriptXMLConsumer):
 def test_can_process_data(mock_write_row: MagicMock,
                           manuscript_consumer: ManuscriptXMLConsumer,
                           manuscript_xml: str):
-    expected = ['1526868166', 'test_file.zip', 'foobar.xml',
+    expected = ['1526868166', 'foobar.xml',
```

Comment:
so this should be `test_file.zip/foobar.xml`, should it not?

Reply:
in order to determine the name of the source file according to the rules in the ticket, the class needs to be initialised with the zip file name. If the zip file is a zip, use `zip/whatever.xml`, if not specified, just use `whatever.xml`. This test doesn't initialise the class properly, using a mock instead. There is no test for the logic in that ticket, I'll add one now.

Reply:
added `test_source_file_names` that goes through the different variations